src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketTestHelper.cs

```diff
@@ -33,6 +33,8 @@ public abstract class SocketHelperBase
         public abstract Task<int> SendAsync(Socket s, ArraySegment<byte> buffer);
         public abstract Task<int> SendAsync(Socket s, IList<ArraySegment<byte>> bufferList);
         public abstract Task<int> SendToAsync(Socket s, ArraySegment<byte> buffer, EndPoint endpoint);
+        public abstract Task SendFileAsync(Socket s, string fileName);
+        public abstract Task SendFileAsync(Socket s, string fileName, ArraySegment<byte> preBuffer, ArraySegment<byte> postBuffer, TransmitFileOptions flags);
```

Comment:
Should these be ReadOnlyMemories instead of the ArraySegment?
The approved API for SendFileAsync is ROM-based, as opposed to e.g. SendAsync which has an overload for ArraySegment.

Reply:
These are wrappers (not the actual socket API-s). We are using `ArraySegment<byte>` so we can also feed the data to sync overloads from the shared test code.

Reply:
It is tricky with `SocketTestHelperMemoryNativeTask`: in that case we copy the data to an actual pointer-based `Memory<byte>` from the ArraySegment in the helper implementation.